src/VisualStudio/Core/Def/Workspace/SourceGeneratedFileManager.cs

```diff
@@ -149,13 +152,27 @@ internal sealed class SourceGeneratedFileManager : IOpenTextBufferEventListener
 
             identity = default;
 
+            // Find GUID in "temporary directory/GUID/path/to/file.cs".
             if (!filePath.StartsWith(_temporaryDirectory))
             {
                 return false;
             }
 
-            var fileInfo = new FileInfo(filePath);
-            return Guid.TryParse(fileInfo.Directory.Name, out var guid) &&
+            var slice = filePath.AsSpan()[_temporaryDirectory.Length..];
+            if (slice.IsEmpty || !PathUtilities.IsDirectorySeparator(slice[0]))
+            {
+                return false;
+            }
+            slice = slice[1..];
+
+            var separatorIndex = slice.IndexOfAny(Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar);
```

Comment:
I would have expected that we already have normalized paths being passed into here so the check for AltDirectorySeperatorChar wasn't necessary, but were you seeing otherwise?